org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -866,6 +868,27 @@ int getVersion(URI uri) {
 		return -1;
 	}
 
+	/**
+ 	 * Submit an asynchronous call (i.e. to the language server) that be inserted into the current position w.r.t.
+ 	 * document change events
+ 	 *
+ 	 * @param <U> Computation return type
+ 	 * @param file Document to serialise on
+ 	 * @param fn Asynchronous computation on the language server
+ 	 * @return Asynchronous result object.
+ 	 */
+ 	public <U> @Nullable CompletableFuture< U> executeOnCurrentVersionAsync(IFile file,
```

Comment:
I like the idea of adding such a method, however the language server don't only work on IFile but instead work on URIs or even IDocument. IMO the signature should replace IFile by one of those types.

Reply:
Also, could this be made package visible only? The less APIs we expose, the better.

Reply:
Done, and changed to URI instead of File